book/src/development/infrastructure/sync.md

```diff
@@ -15,107 +15,82 @@ done in a bi-weekly basis if there's no urgent changes. This is done starting on
+If there should be merge conflicts, resolve them now and commit with the message
+`Merge from rustc`.[^1]
 
-Here is a TL;DR version of the sync process (all the following commands have
-to be run inside the `rust` directory):
-
-1. Clone the [`rust-lang/rust`] repository or make sure it is up-to-date.
-2. Checkout the commit from the latest available nightly. You can get it using
-   `rustup check`.
-3. Sync the changes to the rust-copy of Clippy to your Clippy fork:
-    ```bash
-    # Be sure to either use a net-new branch, e.g. `sync-from-rust`, or delete the branch beforehand
-    # because changes cannot be fast forwarded and you have to run this command again.
-    git subtree push -P src/tools/clippy clippy-local sync-from-rust
-    ```
-
-    > _Note:_ Most of the time you have to create a merge commit in the
-    > `rust-clippy` repo (this has to be done in the Clippy repo, not in the
-    > rust-copy of Clippy):
-    ```bash
-    git fetch upstream  # assuming upstream is the rust-lang/rust remote
-    git checkout sync-from-rust
-    git merge upstream/master --no-ff
-    ```
-    > Note: This is one of the few instances where a merge commit is allowed in
-    > a PR.
-4. Bump the nightly version in the Clippy repository by changing the date in the
-   rust-toolchain file to the current date and committing it with the message:
-   ```bash
-   git commit -m "Bump nightly version -> YYYY-MM-DD"
-   ```
-5. Open a PR to `rust-lang/rust-clippy` and wait for it to get merged (to
-   accelerate the process ping the `@rust-lang/clippy` team in your PR and/or
-   ask them in the [Zulip] stream.)
+> Note: If the version tests fail, refer to [bump version] in the release
+> documentation.
 
+Open a PR to `rust-lang/rust-clippy` and wait for it to get merged (to
+accelerate the process ping the `@rust-lang/clippy` team in your PR and/or ask
+them in the [Zulip] stream.)
+
+[bump version]: release.md#bump-version
 [Zulip]: https://rust-lang.zulipchat.com/#narrow/stream/clippy
 
+[^1]: The message is not really important, but consistency is nice.
+
 ## Performing the sync from Clippy to [`rust-lang/rust`]
 
-All the following commands have to be run inside the `rust` directory.
+The other direction is done by running
+
+```
+$ cargo dev sync push /path/to/rust --user <GitHub-name>
+```
+
+Where the `/path/to/rust` is a relative path to a Rust clone and the
+`<GitHub-name>` is your GitHub user name. This is required for pushing the sync
+to GitHub and opening a PR.
+
+If everything went right, there will be a GitHub link that has to be used to
+open the sync PR in the Rust repository. The PR description must look like this:
+
+```
+Clippy subtree update
+
+r? @ghost
+
+Sync from Clippy commit: rust-lang/rust-clippy@<sha1>
+```
+
+The title must be kept as is, to [tell triagebot] that this is a sync PR.
+
+If you're a Clippy maintainer, you can `r+` the PR yourself. If not, change `r?
+@ghost` to `r? clippy`.
```

Comment:
Remove this. Or rather: Move it to the other sync direction description. No merging yourself in the Rust repo!